pkg/cache/v3/linear.go

```diff
@@ -97,58 +147,63 @@ func WithLogger(log log.Logger) LinearCacheOption {
 // NewLinearCache creates a new cache. See the comments on the struct definition.
 func NewLinearCache(typeURL string, opts ...LinearCacheOption) *LinearCache {
 	out := &LinearCache{
-		typeURL:       typeURL,
-		resources:     make(map[string]types.Resource),
-		watches:       make(map[string]map[int64]ResponseWatch),
-		watchAll:      make(map[int64]ResponseWatch),
-		deltaWatches:  make(map[int64]DeltaResponseWatch),
-		versionMap:    nil,
-		version:       0,
-		versionVector: make(map[string]string),
-		log:           log.NewDefaultLogger(),
+		typeURL:         typeURL,
+		resources:       make(map[string]*cachedResource),
+		resourceWatches: make(map[string]watches),
+		wildcardWatches: newWatches(),
+		version:         0,
+		currentWatchID:  0,
+		log:             log.NewDefaultLogger(),
 	}
 	for _, opt := range opts {
 		opt(out)
 	}
-	for name := range out.resources {
-		out.versionVector[name] = out.getVersion()
+	for name, resource := range out.resources {
+		resource.cacheVersion = out.getVersion()
+		out.resources[name] = resource
 	}
 	return out
 }
 
-func (cache *LinearCache) computeSotwResponse(watch ResponseWatch, ignoreReturnedResources bool) *RawResponse {
+func (cache *LinearCache) computeResourceChange(sub Subscription, ignoreReturnedResources, useStableVersion bool) (updated, removed []string, err error) {
```

Comment:
small suggestion: use `alwaysReturnAllResources` for the first boolean. I find that more clear than refering to "returned resources".

I think this method does enough to warrant a godoc comment explaining what it does.